openedx/features/course_experience/views/course_home.py

```diff
@@ -131,7 +139,9 @@ def render_to_fragment(self, request, course_id=None, **kwargs):
                     request, course_id=course_id, **kwargs
                 )
             course_sock_fragment = CourseSockFragmentView().render_to_fragment(request, course=course, **kwargs)
-            has_visited_course, resume_course_url = self._get_resume_course_info(request, course_id)
+            has_visited_course, resume_course_url = False, None
+            if not request.user.is_anonymous():
+                has_visited_course, resume_course_url = self._get_resume_course_info(request, course_id)
```

Comment:
The multiple checks here are making the code complex quite complex. Try this instead:

1. Move all the assignments to `None` in the `else:` clause above the line `if user_access['is_enrolled'] or user_access['is_staff']:`.
2. Then instead of modifying the `if`, add another clause:
```
else if (course.access is "preview" or course.access is "public"):
  outline_fragment = <init>
  course_sock_fragment = <init>
```

This will make it clear what is being set up in each of the three cases.

Reply:
👍